app/apphandlers/setup.go

```diff
@@ -176,40 +174,34 @@ func monitorDestRouters(ctx context.Context, routerFactory *router.Factory, batc
 	routerFactory.RouterDB.DeleteExecuting()
 	batchRouterFactory.RouterDB.DeleteExecuting()
 
-loop:
-	for {
-		select {
-		case <-ctx.Done():
-			break loop
-		case config := <-ch:
```

Comment:
It looked more correct in the previous version with the `select`. How come you decided to just range over `ch`?
What happens if the context gets canceled and there is no value on `ch`? Won't we be hanging?

Reply:
> I understand that the expectation would be that when `ctx.Done`, `ch` will be closed as well

Yeah I understand that, what I don't get is why we are doing this change. What do we gain by doing it this way beside just deleting a few lines of code? `Subscribe` lives in a different package and a different package might even be in another repo or have bugs we don't know about as far as I'm concerned. So unless we gain something by not honouring the context and hoping somebody else will, I would keep it simple and honour the context.

Reply:
Additionally, the fact that you feel you need a comment on `Subscribe` to document this I think it supports my argument above.

Reply:
I believe this implementation is simpler and safer. Not everyone will no to use a 
```go
select {
		case <-ctx.Done():
			break loop
		case config := <-ch:
}
```

When reading from a channel. I much prefer the default and correct way of using this to be as simple as possible.

Reply:
Btw I was inspired by [etcd API](https://github.com/etcd-io/etcd/blob/client/v3.5.4/client/v3/watch.go#L75) and I preferred all the other approaches.

My other option would be a blocking call with a callback as a parameter.

Reply:
Resource leaking is a valid argument but my expectation generally when I see things like `.Subscribe` is that you would that object like most clients, for example:

```go
client := NewClient()
defer client.Shutdown(shutdownCtx) //  in main.go

ch, err := client.Subscribe(ctx, "something")
if err != nil {
    // ...
}

for {
    select {
        case <-ctx.Done():
            return
        case msg := <-ch:
            // do something with msg
    }
}
```

If there is no "shutdown" then I see your approach as valid to avoid leaking. Either way I think they are both valid approaches :+1: